gtsfm/data_association/data_assoc.py

```diff
@@ -196,6 +202,44 @@ def run(
 
         return connected_data, data_assoc_metrics
 
+    def run_triangulation(
+        self,
+        cameras: Dict[int, gtsfm_types.CAMERA_TYPE],
+        tracks_2d: List[SfmTrack2d],
+    ) -> Tuple[List[Delayed], List[Delayed], List[Delayed]]:
+        """Performs triangulation of 2D tracks in parallel.
+
+        Ref: https://docs.dask.org/en/stable/delayed-best-practices.html#compute-on-lots-of-computation-at-once
+
+        Args:
+            cameras: list of cameras wrapped up as Delayed.
+            tracks_2d: list of tracks wrapped up as Delayed.
+
+        Returns:
+            sfm_tracks: List of triangulated tracks.
+            avg_track_repoj_errors: List of average reprojection errors per track.
+            triangulation_exit_codes: exit codes for each triangulation call.
+        """
+        # Initialize 3D landmark for each track
+        point3d_initializer = Point3dInitializer(cameras, self.triangulation_options)
+
+        # Loop through tracks and and generate delayed triangulation tasks.
+        triangulation_results = []
+        for track_2d in tracks_2d:
+            triangulation_results.append(dask.delayed(point3d_initializer.triangulate)(track_2d))
+
+        # Perform triangulation in parallel.
+        triangulation_results = dask.compute(*triangulation_results)
```

Comment:
do we need the compute? can we not return delayed objects?

Reply:
Returning delayed objects from delayed calls causes problems with Dask's collection API.